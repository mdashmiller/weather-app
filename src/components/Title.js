import React from 'react'
import { Link } from 'react-router-dom'

const Title = () =>
	<Link to='/'>
		<h1 id="landing-title" className="title-gold">
			Weather
			<span className="title-grey">
				Now
			</span>
		</h1>
	</Link>

export default Title
