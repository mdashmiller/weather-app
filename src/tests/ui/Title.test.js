import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { BrowserRouter, Link } from 'react-router-dom'

import Title from '../../components/ui/Title'

Enzyme.configure({ adapter: new Adapter() })

describe('<Title /> rendering', () => {

	it('renders without crashing', () => {
	  	const div = document.createElement('div')
	  	ReactDOM.render(
	  		<BrowserRouter>
	  			<Title />
	  		</BrowserRouter>
	  	, div)
	  	ReactDOM.unmountComponentAtNode(div)
	})

	test('has a valid snapshot', () => {
		const component = renderer.create(
			<BrowserRouter>
				<Title />
			</BrowserRouter>
		)
		let tree = component.toJSON()
		expect(tree).toMatchSnapshot()
	})

	it('renders 1 <Link> component', () => {
		const wrapper = shallow(<Title />)
		expect(wrapper.find(Link).length).toBe(1)
	})

	it('renders 1 <h1> element', () => {
		const wrapper = shallow(<Title />)
		expect(wrapper.find('h1').length).toBe(1)
	})

	it('renders 1 <span> element', () => {
		const wrapper = shallow(<Title />)
		expect(wrapper.find('span').length).toBe(1)
	})

})

describe('<Title /> Link functionality', () => {
	
	it('contains a <Link> to "/"', () => {
		const wrapper = shallow(<Title />)
		expect(wrapper.find(Link).prop('to')).toBe('/')
	})

})
