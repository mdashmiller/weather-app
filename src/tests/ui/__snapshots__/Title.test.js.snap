// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> rendering has a valid snapshot 1`] = `
<a
  href="/"
  onClick={[Function]}
>
  <h1
    className="title-gold"
    id="landing-title"
  >
    Weather
    <span
      className="title-grey"
    >
      Now
    </span>
  </h1>
</a>
`;
